sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,6 +11,190 @@ struct transport_options {
+            if (entrySet) {
+                jobject set = (*env)->CallObjectMethod(env, data, entrySet);
+
+                if (set) {
+                    jclass setClass = (*env)->FindClass(env, "java/util/Set");
+
+                    if (setClass) {
+                        jmethodID iterator =
+                                (*env)->GetMethodID(env, setClass, "iterator", "()Ljava/util/Iterator;");
+
+                        if (iterator) {
+                            jobject iter = (*env)->CallObjectMethod(env, set, iterator);
+
+                            if (iter) {
+                                jclass iteratorClass = (*env)->FindClass(env, "java/util/Iterator");
+
+                                if (iteratorClass) {
+                                    jmethodID hasNext = (*env)->GetMethodID(env, iteratorClass, "hasNext", "()Z");
+
+                                    if (hasNext) {
+                                        jmethodID next =
+                                                (*env)->GetMethodID(env, iteratorClass, "next", "()Ljava/lang/Object;");
+
+                                        if (next) {
+                                            jclass entryClass = (*env)->FindClass(env, "java/util/Map$Entry");
+                                            if (entryClass) {
+                                                jmethodID getKey =
+                                                        (*env)->GetMethodID(env, entryClass, "getKey", "()Ljava/lang/Object;");
+
+                                                jmethodID getValue =(*env)->GetMethodID(env, entryClass, "getValue", "()Ljava/lang/Object;");
+
+                                                // Iterate over the entry Set
+                                                while ((*env)->CallBooleanMethod(env, iter, hasNext)) {
+                                                    jobject entry = (*env)->CallObjectMethod(env, iter, next);
+                                                    jstring key = (jstring) (*env)->CallObjectMethod(env, entry, getKey);
+                                                    // TODO: value can be object e this would make it harder
+                                                    jstring value = (jstring) (*env)->CallObjectMethod(env, entry, getValue);
+                                                    const char* keyStr = (*env)->GetStringUTFChars(env, key, NULL);
+                                                    if (!keyStr) {  // Out of memory
+                                                        return;
+                                                    }
+                                                    const char* valueStr = (*env)->GetStringUTFChars(env, value, NULL);
+                                                    if (!valueStr) {  // Out of memory
+                                                        (*env)->ReleaseStringUTFChars(env, key, keyStr);
```

Comment:
We never called this for any of the other cases. should we?

Reply:
it'd not be a bad idea, but those values are set by us and quite small Strs length, here could be arbitrary data set by customers like a full JSON response, that's why it's so important, good catch.